src/lib/components/identity/identity-details.svelte

```diff
@@ -31,16 +31,27 @@
         revoking = false
     }
 
-    async function handleRoleChange(role: UserRoles): Promise<void> {
-        identity.role = role
-        loading = true
-        caret = false
-        await updateIdentity(identity)
-        loading = false
-        caret=true
+    async function handleRoleChange(userRole: UserRoles): Promise<void> {
+        if (identity.role !== userRole) {
+            loading = true
+            const updateDid = identity
```

Comment:
This won't create a copy of the identity object but just a reference. So changing `updateDid` will also change `identity`.